### ISSUE TITLE
Feature request: Compare new Lambda function name against regex expression

### ISSUE THREAD

**Issue by erikpaasonen** [wata727/tflint#218]
There's an opportunity to validate an `aws_lambda_function` resource's `function_name` parameter. If the name contains certain characters such as `.` the Terraform Apply step will fail on a validation check. Thankfully the error message itself provides the regex expression that AWS is using to validate. Would be great to catch this further upstream.

### Sample Terraform config file
``` hcl
resource "aws_lambda_function" "test_lambda" {
  filename         = "lambda_function_payload.zip"
  function_name    = "aws.lambda_function_name"
  role             = "${aws_iam_role.iam_for_lambda.arn}"
  handler          = "exports.test"
  source_code_hash = "${base64sha256(file("lambda_function_payload.zip"))}"
  runtime          = "nodejs4.3"
}
```

### Proposed behavior
Tflint could output some sort of error when this happens.
```
$ tflint --deep
template.tf
        ERROR:3 "aws.lambda_function_name" is invalid function name. (aws_lambda_invalid_function_name)

Result: 1 issues  (1 errors , 0 warnings , 0 notices)
```

### Actual behavior
Tflint --deep passes:
```
$ tflint --deep
Awesome! Your code is following the best practices :)
```

But then upon Terraform Apply, the apply fails with the following error message:
```
Error: Error applying plan:

1 error(s) occurred:

* aws_lambda_function.test_lambda: 1 error(s) occurred:

* aws_lambda_function.test_lambda: Error creating Lambda function: ValidationException: 1 validation error detected: Value 'aws.lambda_function_name' at 'functionName' failed to satisfy constraint: Member must satisfy regular expression pattern: (arn:(aws[a-zA-Z-]*)?:lambda:)?([a-z]{2}((-gov)|(-iso(b?)))?-[a-z]+-\\d{1}:)?(\\d{12}:)?(function:)?([a-zA-Z0-9-_]+)(:(\\$LATEST|[a-zA-Z0-9-_]+))?
\tstatus code: 400, request id: b9c3aa74-9407-11e8-8cba-37b34d40010a

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```
Most of the regex expression appears to be checking for the possibility that the string is formatted like an AWS ARN. Here's the key part of the regex expression that the bare function name itself needs to match:
``` regex
([a-zA-Z0-9-_]+)
```

**Comment by wata727** [comment]
I like this proposal. I'm going to work on this after core refactoring. If you have any patches, I will review it :)

**Comment by wata727** [comment]
Added [aws_lambda_function_invalid_function_name](https://github.com/wata727/tflint/blob/v0.9.0/rules/awsrules/models/aws_lambda_function_invalid_function_name.go) rule in v0.9.0.
